test/integration/authorization_test.go

```diff
@@ -282,8 +282,15 @@ func TestAuthorizationResolution(t *testing.T) {
 		t.Fatalf("unexpected error: %v", err)
 	}
 
-	// try to add Valerie to a non-existent role
-	if err := addValerie.AddRole(); !kapierror.IsNotFound(err) {
+	// try to add Valerie to a non-existent role, looping until it is true due to
+	// the policy cache taking time to react
+	if err := wait.Poll(time.Second, 2*time.Minute, func() (bool, error) {
```

Comment:
no, this is waiting for the cache to reflect the role deletion above (before it would have failed the live check immediately). 2 minutes seems egregious, though